workshop/content/intermediate/templates/dynamic-references/_index.md

```diff
@@ -0,0 +1,238 @@
+
+```shell
+aws lambda invoke --function-name FUNCTION_NAME /dev/stdout
+
+"Attempting to connect to database db.us-east-1.rds.amazonaws.com:3306"
+```
+
+Congratulations! You learned how to use dynamic references with AWS Secrets Manager.
+
+### Challenge
+
+In this exercise, you will reinforce your understanding of *dynamic references.*
+
+AWS Lambda supports specifying memory configuration for a function with the `MemorySize` property. Your task is to create a Parameter Store parameter with the AWS CLI, where you set the memory size to use for a Lambda function that you will describe in the `lambda_memory_size.yaml` template. You will then create a dynamic reference to version `1` of the parameter you created.
+
+{{%expand "Need a hint?" %}}
+
+* Review the CloudFormation [User Guide](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-memorysize) to understand how to specify the `MemorySize` configuration for a Lambda Function.
+* Review the CloudFormation [User Guide](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-ssm-pattern) for help with constructing a dynamic reference string to a specific version of  an SSM parameter.
+
+{{% /expand %}}
+
+{{%expand "Want to see the solution?" %}}
+
+* Create an SSM parameter specifying the required Memory Configuration using the following command:
+
+```shell
+aws ssm put-parameter --name "/lambda/memory-size" \
+                      --value "256" \
+                      --type "String" \
+                      --region "us-east-1"
+```
+
+* Navigate to the `code/workspace/dynamic-references` directory. Update the template by appending the `Properties` section with the `MemorySize` property using a dynamic reference to the parameter as such:
+
+```yaml
+  HelloWorldFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Role: !GetAtt FunctionExecutionRole.Arn
+      Handler: index.handler
+      Runtime: python3.7
+      MemorySize: '{{resolve:ssm:/lambda/memory-size:1}}'
+      Code:
+        ZipFile: |
+          import os
+
+          def handler(event, context):
+              return "Hello World!"
+```
+
+You can find the full solution at `code/solutions/dynamic-references/lambda_memory_size.yaml` example template.
+
+{{% /expand %}}
+
+### Cleanup
+
+1. In the CloudFormation console, select the stack you have created in this lab. For example: `cfn-workshop-ec2-stack`.
+2. Choose **Delete** to delete the stack, and then choose **Delete stack** to confirm.
+3. Repeat steps above for stacks you created with this lab, for example: `cfn-workshop-database-stack`, `cfn-workshop-lambda-stack`.
+4. Delete the SSM parameters you created to store the AMI ID and `MemorySize` configuration using the following command:
+
+```yaml
+aws ssm delete-parameters --names "/golden-images/amazon-linux-2" \
+                                  "/lambda/memory-size"
+```
+
```

Comment:
Also, please add instructions on deleting CloudWatch Logs resource(s) created with this lab.